subsys/bluetooth/services/ots/dir_list/ots_dir_list_lazy.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/dlist.h>
+#include <sys/byteorder.h>
+
+#include <bluetooth/services/ots.h>
+#include "../ots_internal.h"
+#include "../ots_obj_manager_internal.h"
+#include "ots_dir_list_internal.h"
+#include "ots_dir_list_impl_internal.h"
+
+#include <logging/log.h>
+
+LOG_MODULE_DECLARE(bt_ots, CONFIG_BT_OTS_LOG_LEVEL);
+
+/* In order to transmit the maximum size we choose CONFIG_BT_L2CAP_TX_MTU
```

Comment:
`ots_l2cap.c` uses `BT_L2CAP_SDU_BUF_SIZE(CONFIG_BT_OTS_L2CAP_CHAN_TX_MTU)` as the buffer size, so I suppose that's would be better to use, right?

Reply:
I think you are right but we should only use `CONFIG_BT_OTS_L2CAP_CHAN_TX_MTU` and not `BT_L2CAP_SDU_BUF_SIZE(CONFIG_BT_OTS_L2CAP_CHAN_TX_MTU)`. I don't believe we need to allocate the SDU header room.

Reply:
> I don't believe we need to allocate the SDU header room.

I guess that depends :) If we don't then the MTU will not represent the size of data that can be sent. I think it makes more sense to use `BT_L2CAP_SDU_BUF_SIZE` so that a MTU of e.g. 50 means that we can actually send 50 bytes of data in a single TX.

Reply:
If I understand it correctly the OTS L2CAP layer at the moment isn't a zero copy implementation. It reserves its own TX buffer in `ots_l2cap.c` as you had mentioned:
```c
NET_BUF_POOL_FIXED_DEFINE(ot_chan_tx_pool, 1,
			  BT_L2CAP_SDU_BUF_SIZE(CONFIG_BT_OTS_L2CAP_CHAN_TX_MTU), 8,
			  NULL);
```
But later the data provided by an OTS application is copied into this buffer with consideration for the SDU header:
```c
	/* Prepare buffer for sending. */
	buf = net_buf_alloc(&ot_chan_tx_pool, K_FOREVER);
	net_buf_reserve(buf, BT_L2CAP_SDU_CHAN_SEND_RESERVE);
	net_buf_add_mem(buf, &l2cap_ctx->tx.data[l2cap_ctx->tx.len_sent], len);
```
So I don't think we need to account for the SDU.